test/source/handler.test.js

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2025 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+/* eslint-env mocha */
+/* eslint-disable no-param-reassign */
+import assert from 'assert';
+import { promisify } from 'util';
+import zlib from 'zlib';
+import { Headers, Request } from '@adobe/fetch';
+import { AuthInfo } from '../../src/auth/auth-info.js';
+import { main } from '../../src/index.js';
+import { Nock, SITE_CONFIG } from '../utils.js';
+
+const gunzip = promisify(zlib.gunzip);
+
+function assertHeadersInclude(headers, expected) {
+  const actual = headers.plain();
+  Object.entries(expected).forEach(([key, value]) => {
+    assert.strictEqual(actual[key], value, `Expected header ${key} to be ${value}`);
+  });
+}
+
+describe('Source Handler Tests', () => {
+  /** @type {import('../utils.js').NockEnv} */
+  let nock;
+
+  beforeEach(() => {
+    nock = new Nock().env();
+    nock.siteConfig(SITE_CONFIG);
+  });
+
+  afterEach(() => {
+    nock.done();
+  });
+
+  function setupContext(suffix, { attributes } = {}) {
+    return {
+      attributes: {
+        authInfo: new AuthInfo().withAuthenticated(true),
+        ...attributes,
+      },
+      env: {
+        HELIX_STORAGE_DISABLE_R2: 'true',
+      },
+      pathInfo: { suffix },
+    };
+  }
+
+  it('handles GET requests', async () => {
+    nock.source()
+      .getObject('/org/site/hello.html')
+      .reply(200, Buffer.from('<body>Hello, world!</body>'), {
+        'content-type': 'text/html',
+        'content-length': '26',
+        'last-modified': new Date(946684800000).toUTCString(), // RFC HTTP date format!
```

Comment:
we use ISO everywhere else

Reply:
yes, but this is _our_ json response, and we can put in what we want :-) 
I prefer the ISO, as it is easier to parse and has less redundant information

Reply:
ah right... forget was I said... must be the jet-lag and brain fog :-)